python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1263,28 +1249,24 @@ def execute_asset_backfill_iteration_inner(
         failed_and_downstream_subset = AssetGraphSubset()
         next_latest_storage_id = instance_queryer.instance.event_log_storage.get_maximum_record_id()
     else:
-        updated_materialized_subset_data = None
-        for (
-            updated_materialized_subset_data
-        ) in get_asset_backfill_iteration_materialized_partitions(
+        next_latest_storage_id = instance_queryer.instance.event_log_storage.get_maximum_record_id()
+        # Events are not guaranteed to be written to the event log in monotonic increasing order,
+        # so we wait a few seconds to ensure all events up until next_latest_storage_id have been
+        # written.
+        time.sleep(3)
```

Comment:
Can we make this an env var so that we can tweak if needed?

Reply:
yup -- changed